### ISSUE TITLE
"feast apply" on command line for GCP provider does not use active gcloud service account

### ISSUE THREAD

**Issue by dnfrantz7** [feast-dev/feast#1540]
## Expected Behavior
when running "feast apply" locally for a GCP project from command line, a separate service account with proper permissions can be configured and utilized. This account would be separate from the developer's personal account / credentials.

For this example my personal account is named david@domain.com and the service account for feast is named feast-dev so the full IAM name is feast-dev@my-gcp-project.iam.gserviceaccount.com

## Current Behavior
running `feast apply` uses default application account, even when `gcloud auth list` shows that another service account is currently activated.

## Steps to reproduce
* add new service account -> `gcloud auth activate-service-account --key-file=<path-to-json-credentials>`
* to show active account is set to new service-account -> `gcloud auth list`
* if not set then use `gcloud config set account feast-dev@my-gcp-project.iam.gserviceaccount.com` to set active account to service account
* run `feast apply` in test repository

test repository feast_storage.yaml example:
```
project: dev
registry: gs://my-feast-bucket/registry.db
provider: gcp
online_store:
  type: datastore
  project_id: my-gcp-project
```

`feast apply` will error because incorrect gcloud account was utilized:
google.api_core.exceptions.Forbidden: 403 GET https://storage.googleapis.com/storage/v1/b/my-feast-bucket?projection=noAcl&prettyPrint=false: david@domain.com does not have storage.buckets.get access to the Google Cloud Storage bucket.

### Specifications

- Version: Feast 0.10.3
- Platform: MacOS Big Sur 11.2.3

## Possible Solution
Perhaps usage of `gcloud auth application-default login` -> https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login is interfering with which account is considered to be activated when commands are run.

**Comment by woop** [comment]
Thanks @dnfrantz7 I'm having a look right now.

**Comment by woop** [comment]
@dnfrantz7 

I think I've found the solution, but I'd like you to confirm. 

I ran the `google/cloud-sdk` docker image, installed Python 3.9, then installed the master branch of Feast. I created a new repo and set up my configuration as yours. I think set up a service account using the same commands as you (and then created a bucket using that service account using the command line).

I then ran `feast apply`, which gave me 
```
    raise ValueError("Anonymous credentials cannot be refreshed.")
ValueError: Anonymous credentials cannot be refreshed.
```

After removing a try/catch that was hiding an exception (over here https://github.com/feast-dev/feast/pull/1561/files) I was able to see that my `GOOGLE_APPLICATION_CREDENTIALS` needed to be set.

```
google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application. For more information, please see https://cloud.google.com/docs/authentication/getting-started
```

Running `GOOGLE_APPLICATION_CREDENTIALS=/feast/key.json feast apply` resolved the issue.
